packages/gasket-plugin-webpack/lib/init-webpack.js

```diff
@@ -30,22 +24,18 @@ function setupContext(gasket, context, name) {
 module.exports = function initWebpack(gasket, initConfig, context) {
   const WebpackMetricsPlugin = require('./webpack-metrics-plugin');
 
-  const baseConfig = {
+  let baseConfig = {
     ...initConfig,
     plugins: [
       ...(initConfig && initConfig.plugins ? initConfig.plugins : []),
       new WebpackMetricsPlugin({ gasket })
     ].filter(Boolean)
   };
 
-  // TODO: Remove in next major version
-  let mergedConfig = require('./deprecated-merges')(gasket, baseConfig, context);
-
   // eslint-disable-next-line no-sync
-  gasket.execApplySync('webpackConfig', (plugin, handler) => {
-    const name = plugin ? plugin.name || 'unnamed plugin' : 'app lifecycle';
-    mergedConfig = handler(mergedConfig, setupContext(gasket, context, name));
+  gasket.execApplySync('webpackConfig', (_, handler) => {
+    baseConfig = handler(baseConfig, setupContext(gasket, context));
```

Comment:
Because we don't need to pick out the plugin name for the deprecation warning, I think we should be able to simply this to use `execWaterfallSync` like:
```suggestion
return gasket.execWaterfallSync('webpackConfig', baseConfig, setupContext(context));
```